scripts/globals/mobskills.lua

```diff
@@ -265,59 +265,58 @@ end
-    end
+    local finalDamage = 0
 
-    -- plus 100 forces it to be a number
-    local mab = (100 + mob:getMod(xi.mod.MATT)) / (100 + target:getMod(xi.mod.MDEF) + mdefBarBonus)
-    mab = utils.clamp(mab, 0.7, 1.3)
-
-    if tpeffect == xi.mobskills.magicalTpBonus.DMG_BONUS then
-        damage = damage * (((skill:getTP() / 10) * tpvalue) / 100)
+    -- Base damage
+    if tpEffect == xi.mobskills.magicalTpBonus.DMG_BONUS then
+        finalDamage = math.floor(baseDamage * action:getTP() * tpMultiplier / 1000)
     end
 
-    -- resistance is added last
-    local finaldmg = damage * mab * dmgmod
-
-    -- get resistance
+    -- Get bonus macc.
     local petAccBonus = 0
-    if mob:isPet() and mob:getMaster() ~= nil then
-        local master = mob:getMaster()
-        if mob:isAvatar() then
-            petAccBonus = utils.clamp(master:getSkillLevel(xi.skill.SUMMONING_MAGIC) - master:getMaxSkillLevel(mob:getMainLvl(), xi.job.SMN, xi.skill.SUMMONING_MAGIC), 0, 200)
+    if actor:isPet() and actor:getMaster() ~= nil then
+        local master = actor:getMaster()
+        if actor:isAvatar() then
+            petAccBonus = utils.clamp(master:getSkillLevel(xi.skill.SUMMONING_MAGIC) - master:getMaxSkillLevel(actor:getMainLvl(), xi.job.SMN, xi.skill.SUMMONING_MAGIC), 0, 200)
         end
 
-        local skillchainTier, _ = xi.magicburst.formMagicBurst(element, target)
+        local skillchainTier, _ = xi.magicburst.formMagicBurst(actionElement, target)
         if
-            mob:getPetID() > 0 and
+            actor:getPetID() > 0 and
             skillchainTier > 0
         then
             petAccBonus = petAccBonus + 25
         end
     end
 
-    local resist       = xi.mobskills.applyPlayerResistance(mob, nil, target, mob:getStat(xi.mod.INT)-target:getStat(xi.mod.INT), petAccBonus, element)
-    local magicDefense = getElementalDamageReduction(target, element)
+    -- Multipliers.
+    local sdt                         = xi.spells.damage.calculateSDT(target, actionElement)
+    local resist                      = xi.mobskills.applyPlayerResistance(actor, nil, target, actor:getStat(xi.mod.INT) - target:getStat(xi.mod.INT), petAccBonus, actionElement)
+    local dayAndWeather               = xi.spells.damage.calculateDayAndWeather(actor, 0, actionElement)
+    local magicBonusDiff              = xi.spells.damage.calculateMagicBonusDiff(actor, target, 0, 0, actionElement)
+    local targetMagicDamageAdjustment = xi.spells.damage.calculateTMDA(target, actionElement)
 
-    finaldmg       = finaldmg * resist * magicDefense
-    returninfo.dmg = finaldmg
+    -- Calculate final damage.
+    finalDamage = math.floor(finalDamage * sdt)
+    finalDamage = math.floor(finalDamage * resist)
+    finalDamage = math.floor(finalDamage * dayAndWeather)
+    finalDamage = math.floor(finalDamage * magicBonusDiff)
+    finalDamage = math.floor(finalDamage * targetMagicDamageAdjustment)
+    finalDamage = math.floor(finalDamage * damageModifier)
```

Comment:
It looks like (I may have missed one) all these damage modifiers are 1? Do we want to just remove this completely. Not even sure what this is

Reply:
I havent checked all 250+ scripts that use this modifier

Reply:
The PR is already big enough as it is already though. If it turns out to always be 1, I'll remove it in a different PR.

For now, the main focus was to fix the pressing issue: MAB and TMDA being applied twice while, at the same time, TMDA and weather multipliers not being used for most magical mobskills